### ISSUE TITLE
CAST + alias + IS_NULL/IS_NOT_NULL filter expression does not mix

### ISSUE THREAD

**Issue by egalpin** [apache/pinot#11139]
```sql
SELECT CAST(myCol AS LONG) AS myCol WHERE myCol IS NOT NULL
```

The above will fail with 

```
Caused by: java.lang.UnsupportedOperationException: Unsupported predicate type: IS_NULL
    at org.apache.pinot.core.operator.filter.predicate.PredicateEvaluatorProvider.getPredicateEvaluator(PredicateEvaluatorProvider.java:89)
```

Is that because there is no `IsNotNullPredicateEvaluatorFactory` akin to the other factories found in `PredicateEvaluatorProvider`.  Is having `Is[Not]NullPredicateEvaluatorFactory` a bad idea, or just something that has not yet been implemented?

The above query works just fine in the case where the alias is changed to be distinct from the original column name (or alias removed entirely). This works fine:

```sql
SELECT CAST(myCol AS LONG) AS __myCol WHERE myCol IS NOT NULL
```

**Comment by Jackie-Jiang** [comment]
Semantic wise, I feel the intention is to access the original `myCol` instead of the alias. Pinot will actually replace the column with `CAST(myCol AS LONG)`.

So there are 2 problems:
1. Column in WHERE clause should not be treated as alias because it is not allowed to access alias in WHERE clause
2. Currently we cannot do NULL filter on transformed column. @shenyu0127 is working on filter NULL support, and should be able to support this

Seems using an existing column name as alias is not best practice because that can cause confusion

**Comment by Jackie-Jiang** [comment]
Alias should not be applied to filter. Fixed in #11610
